### PR TITLE
feat: add PWA feats & Cypress tests closes #157

### LINKED ISSUE #157: Service worker
Would adding a service worker js be useful to make the app offline capable? :thinking: 

Pros :+1: 
- Makes app work offline :sunglasses: 

Cons :-1: 
- Adds little more complexity to code :hammer_and_wrench:  :computer: 
- Not supported by all browsers :frowning_face: 
- May not notice much difference if developing on local host 

If you think it's worth adding, happy to make PR :smile:

### PR DESCRIPTION
**Pull request by fozy81**
Service worker, Cypress tests, PWA features (manifest.json)
- let's see if the tests pass! - had problems with the 'sign-out' test
- may want to re-deploy demo tracker app

### CODE REVIEW

**Comment by fozy81** [comment]
Failed test on: 1) should allow me to sign out 

skipping that test for now...and have re-committed changes to master

**Comment by fozy81** [comment]
take two - needed to delete items before running some tests. tests passed locally...fingers-crossed.

**Comment by fozy81** [comment]
okay, cypress tests passing - but some issue with deploying script?

**Comment by gr2m** [comment]
The tests are still assuming they are run with selenium, hence the two builds, one configuring to use saucelabs, one to use selenium. 

I think we can get rid of the two builds, I’d just start Hoodie in memory and run tests with Cypress?

I’m not sure what you mean with "issue with deploying script"?

**Comment by fozy81** [comment]
> The tests are still assuming they are run with selenium, hence the two builds, one configuring to use saucelabs, one to use selenium.
> I think we can get rid of the two builds, I’d just start Hoodie in memory and run tests with Cypress?

I've removed the two builds + started Hoodie in memory. :+1: 

> I’m not sure what you mean with "issue with deploying script"?

There's a docker [deploy script in the travis.yml](https://github.com/hoodiehq/hoodie-app-tracker/blob/563d55d30fd2fd36953fda8711170d223c8dff24/.travis.yml#L28) which pushes to hoodiehq/hoodie-app-tracker? I think this is causing this 'HEAD detached' [travis failure](https://travis-ci.org/fozy81/hoodie-app-tracker#L4031)?

I'm not very familiar with travis/docker so all trail and error for me... :wrench: :hammer:

**Comment by gr2m** [comment]
@christophwitzko could you have a look at it please? ^^

**Comment by fozy81** [comment]
@gr2m @christophwitzko  Tests passing! I followed the the instructions on Travis, adding 'skip_cleanup: true'. I think the issue was that running the tests with Cypress added temporary files to the repo and then these files made the repo out of sync with the commit so docker throws an error. see https://docs.travis-ci.com/user/deployment/#Uploading-Files-and-skip_cleanup

**Comment by gr2m** [comment]
I’ll try to have another look this weekend 🙏 sorry for the delay!

File: .travis.yml [hunk 1]
```diff
@@ -19,6 +19,7 @@ before_script:
   # required for headless firefox / selenium test
   - export DISPLAY=:99.0
   - sh -e /etc/init.d/xvfb start
+<<<<<<< HEAD
```
Comment (gr2m) [inline .travis.yml]: looks like a merge conflict?

**Comment by gr2m** [comment]
w00p thanks for getting it all done <3 Let me know if it’s ready from your side!

**Comment by fozy81** [comment]
Finished. My first merge conflict resolved! 😅

**Comment by gr2m** [comment]
w00p congratultations, looks great 👍 thanks for putting all the work in, great to have a more stable testing setup again

**Pull request merged**